include/envoy/common/platform.h

```diff
@@ -61,6 +61,7 @@ typedef ptrdiff_t ssize_t;
 typedef uint32_t mode_t;
 
 typedef SOCKET os_fd_t;
+using os_h_t = HANDLE;
```

Comment:
The os_h_ is a somewhat awkward name imo, at least 'fd' is a well recognized Posix construct. At minimum, os_handle_t, but that's a foreign concept to Posix.

What I'm wondering, because a SOCKET is simply a special case of a HANDLE, is whether we need two types different types for Windows, or whether we can simply continue to use os_fd_t?

In any case, I don't think we want to be jumping between the typedef and using constructions.

Reply:
1. For the `using` vs `typdef` I tried this because `clang-tidy` raises an error with typedef. The error is prefer `using` over typedef. I tried using as an alternative but this did not fix the issue.

1. I can change it to `os_handle_t `. From what I understood while I was reading about these two types I could not find a way to have them as a common abstraction. At least without some extra assumptions or some 'casting'. The reason I went for two different types is because the use of `os_handle_t` is limited to the filesystem implementation.

Reply:
@wrowe this is the clang-tidy error. Do you know what a good resolution would be?

Reply:
@htuch or @mattklein123 any thoughts on using vs typedef in this context?

Reply:
AIUI they are #defined together on windows. Since every socket is a handle, every linux socket is an fd, I'd rather have a resolution at os_fd_t vs os_socket_fd_t if you want two distinct types.

Reply:
There is typically a way via NOLINT to stop tidy complaining. See other examples.

Reply:
Both of the issues are fixed in 47371422958b974e7589af88e931c08f9822c8e1